spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/actions/BaseSparkAction.java

```diff
@@ -253,6 +257,39 @@ protected DeleteSummary deleteFiles(
     return summary;
   }
 
+  protected DeleteSummary deleteFiles(
+      Consumer<Iterable<String>> bulkDeleteFunc, Iterator<FileInfo> files) {
+    DeleteSummary summary = new DeleteSummary();
+    Iterator<List<FileInfo>> groupedIterator = Iterators.partition(files, DELETE_GROUP_SIZE);
+
+    Tasks.foreach(groupedIterator)
+        .suppressFailureWhenFinished()
+        .run(
+            fileList -> {
+              Map<String, List<FileInfo>> filesByType =
```

Comment:
What about adding a private method to handle this? I feel it would help with formatting (if the call fits on a single line). Spotless formats these closures in a weird way.

```
Iterator<List<FileInfo>> fileGroups = Iterators.partition(files, DELETE_GROUP_SIZE);

Tasks.foreach(fileGroups)
    .noRetry()
    .suppressFailureWhenFinished()
    .run(fileGroup -> deleteFileGroup(...));

LOG.info("Deleted {} total files with bulk deletes", summary.totalFilesCount());

return summary;
```